### PR TITLE
Lots of minor tidy-ups to hw/Makefile

### PR DESCRIPTION
**Pull request by rswarbrick**
This isn't a big deal per se, but I've been regenerating registers a lot recently and poring through the console output to find the stuff I cared about was getting a bit painful. I've split the changes into a series of self-contained commits: this might be a bit easier to review but, more importantly, is easier to bisect if I've messed something up.

The point of all of this is to tidy up the output. Before, `make -C hw -j4 all` generated 475 lines. Now it generates 80.

### CODE REVIEW

File: hw/Makefile [hunk 1]
```diff
@@ -130,17 +130,15 @@ $(ips_reg_rust_install): $(tock_rust_destdir)/%: $(REG_OUTPUT_SW_DIR)/% | $(tock
...
-
-$(tops_reg):
-	$(eval $@_TOP := $(strip $(foreach top,$(TOPS),$(findstring $(top),$@))))
-	mkdir -p ${REG_OUTPUT_DV_DIR}/$($@_TOP)
-	${PRJ_DIR}/util/topgen.py -t ${PRJ_DIR}/hw/$($@_TOP)/data/$($@_TOP).hjson \
-		-r -o ${REG_OUTPUT_DV_DIR}/$($@_TOP) ${toolflags}
+$(tops_gen): %_gen:
+	${PRJ_DIR}/util/topgen.py -t $(top-hjson) -o ${PRJ_DIR}/hw/$*/ ${toolflags}
+
+$(tops_reg): %_reg:
```
Comment (tjaychen) [inline hw/Makefile:138]: i'm a bit confused about this usage, does the `%_reg` mean even the ip reg targets will fall into this rule?

Comment (tjaychen) [reply hw/Makefile:138]: ah..is it [this](https://www.gnu.org/software/make/manual/make.html#Static-Pattern)? I think my makefile knowledge just sucks.

File: hw/Makefile [hunk 2]
```diff
@@ -97,11 +97,14 @@ blk-hjson      = $(PRJ_DIR)/hw/ip/$*/$(dir_hjson)/$*.hjson
...
 # require all blocks to use a consistent naming. This should be hooked
 # into ipgen (see #5636) for completeness.
-$(ips_reg): %_reg: otp-mmap lc-state-enc | $(REG_OUTPUT_DV_DIR)
+$(ips_reg): %_reg: lc-state-enc | $(REG_OUTPUT_DV_DIR)
 	$(if $(wildcard $(blk-gen-script)),$(blk-gen-script))
 	${PRJ_DIR}/util/regtool.py ${toolflags} -r $(blk-hjson)
 	${PRJ_DIR}/util/regtool.py -s -t $(REG_OUTPUT_DV_DIR) $(blk-hjson)
 
+# Register generation for otp_ctrl also depends on running gen-otp-mmap.py
+.PHONY: otp-mmap
```
Comment (tjaychen) [inline hw/Makefile:105]: is it generally good practice to declare `PHONY` where its needed and not slap them into one line?
this does makes sense to me..but wanted to double check.

**Comment by tjaychen** [comment]
thanks @rswarbrick ! this generally looks good to me (definitely needed to do some makefile syntax googling).
I'm not an expert on Makefiles though, so maybe others can chime in on best practices etc.

In general everything here makes sense, and thanks for cleaning it up.

File: hw/Makefile [hunk 3]
```diff
@@ -4,7 +4,6 @@
 
 CUR_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 PRJ_DIR := $(realpath ${CUR_DIR}/../)
-export PRJ_DIR
```
Comment (imphil) [inline hw/Makefile:7]: Also, it's typically called REPO_TOP :-)

**Review by imphil — APPROVED** [review]
LGTM, thanks Rupert!

File: hw/Makefile (hunk 1 above)
Comment (rswarbrick) [reply hw/Makefile:138]: Yep, it's a static pattern. They're really neat (and much less confusing than implicit rules for exactly that reason!)

**Pull request merged**